### PR TITLE
Overhaul complex fluid solver

### PR DESCRIPTION
**Pull request by abarret**
There are two commits here. This first commit adds in some missing options for the solver, such as flexibility in choice of convective operator and the ability to specify a function to apply before the divergence of the stress is calculated.

The second commit is an overhaul of the complex fluid solver that enables greater flexibility in specifying viscoelastic models. The major change in this commit is the replacement of `CFRelaxationOperator` with `CFStrategy`, which allows users to specify a relaxation function as well as a conversion function from the conformation tensor to the stress. In particular, this change allows the use of finite extensibility models (e.g. FENE-P) and multiphase models. The second commit also greatly expands the documentation and changes variable names to be consistent with complex fluid literature.

While the second commit changes the API, if people are using the basic complex fluid solver, the set-up is exactly the same. In particular, the only change in the examples is minimal.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Review by drwells — CHANGES_REQUESTED** [review]
I don't know a lot about complex fluids but this looks pretty reasonable. We still need a changelog entry describing the incompatibilities.

File: include/ibamr/CFStrategy.h [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+/*!
+ * Return the conformation tensor, given the evolved data and the evolved type.
+ */
+inline IBTK::MatrixNd
+convert_to_conformation_tensor(const SAMRAI::pdat::CellData<NDIM, double>& C_data,
+                               const SAMRAI::pdat::CellIndex<NDIM>& idx,
+                               TensorEvolutionType evolve_type)
+{
+    IBTK::MatrixNd mat;
```
[the remark this answers is not in the document]
Comment (abarret) [reply include/ibamr/CFStrategy.h:132]: The following dozen lines initialize the full matrix, not just the upper triangular part.

**Comment by abarret** [comment]
> Any experts in complex fluids want to chime in? If not lets merge this by the end of the week.

@colegruninger97 ?

**Comment by colegruninger97** [comment]
> > Any experts in complex fluids want to chime in? If not lets merge this by the end of the week.
> 
> @colegruninger97 ?

Looks great!

**Pull request merged**